stl/inc/experimental/random

```diff
@@ -0,0 +1,63 @@
+// random experimental header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _EXPERIMENTAL_RANDOM_
+#define _EXPERIMENTAL_RANDOM_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#include <random>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+namespace experimental {
+    inline namespace fundamentals_v3 {
+
+        _NODISCARD inline default_random_engine& _Engine() {
+            thread_local default_random_engine _Eng{random_device{}()};
+            return _Eng;
+        }
+
+        template <class _IntType>
+        _NODISCARD _IntType randint(const _IntType _Min, const _IntType _Max) {
+            _RNG_REQUIRE_INTTYPE(randint(), _IntType);
+            _STL_ASSERT(_Min <= _Max, "invalid min and max arguments for randint()");
+            return uniform_int_distribution<_IntType>{_Min, _Max}(_Engine());
+        }
+
+        inline void reseed() {
+            constexpr size_t _N = mt19937::state_size;

```

Comment:
Would `static_assert(is_same_v<default_random_engine, mt19937>, "This code assumes that default_random_engine is mt19937");` be a good idea here? We don't always comment MSVC platform assumptions, but it seems probable that someone may attempt to port this code to libc++ or elsewhere.

Reply:
Added it.

Reply:
FYI, I sent an LWG issue to remove the `default_random_engine` restriction (@BillyONeal reported that `default_random_engine` cannot be changed to other types without breaking ABI):

Modify its specification in 10.1.2.1 [rand.util.randint]/6 accordingly:

<pre><code>void reseed();
void reseed(<s>default_random_engine::result_type</s><ins>uint_fast32_t</ins> value);
</code></pre>


Modify 10.1.2.1 [rand.util.randint]/1 as follows:

A separate *per-thread* engine of <s>type default_random_engine (C++17 §29.6.5)</s><ins>unspecified type that meets the requirements of random number engine (C++17 [rand.req.eng])</ins>, initialized to an unpredictable state, shall be maintained for each thread.<ins>[Note: The implementation may choose the engine type on the basis of performance, size, quality, or any combination of such factors, so as to provide at least acceptable engine behavior for relatively casual, inexpert, and/or lightweight use. -- end note]</ins>

Reply:
So provided this PR is still going forward, would this be applied preemptively and am I correct in thinking these would be the changes?
```diff
-static_assert(
-    is_same_v<default_random_engine, mt19937>, "This code assumes that default_random_engine is mt19937");

- _NODISCARD inline default_random_engine& _Engine() {
+_NODISCARD inline mt19937& _Engine() {

-void reseed(default_random_engine::result_type _Value);
+void reseed(uint_fast32_t _Value);
```